models/classes/class.ItemsService.php

```diff
@@ -187,7 +189,9 @@ public function hasItemContent(core_kernel_classes_Resource $item, $lang = '')
             return false;
         }
 
-        return $this->getItemDirectory($item, $lang)->getFile(QtiFile::FILE)->exists();
+        $file = $this->getItemDirectory($item, $lang)->getFile(QtiFile::FILE);
+
+        return $file->exists() ? !(strpos($file->read(), self::DIV_CLASS_EMPTY) !== false) : false;
```

Comment:
I understood the solution, but do you know how often or what is the risk FE might change the HTML format?

Reply:
i have tackled this by checking the #IMS  channel , 
the FE did not change the template for a long time :) so no issues there
ofc there is always a risk for that to happen but nothing will break the system will happen only the preview will fail in specific conditions

Reply:
I think this validation is really important and now it is disguised inside a `getRootClass` method. Do not you think we should have an specific Service/method::isEmpty to handle this rule? So it can be reused and centralized. Whyt? I am afraid we will need this in other places and probably we will repeat this code.